### ISSUE TITLE
[PAY] Image infinite loading when offline

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#13826]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Turn on the Airplane mode, open the app
2. Open any report that has image, click on image.
3. Notice that the image is infinite loading.

## Expected Result:
 Image should show error or placeholder error.

## Actual Result:
The image is infinite loading.

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?

- [ ] Android / native
- [ ] iOS / native

**Version Number:** 1.2.43-1
**Reproducible in staging?:** y
**Reproducible in production?:** y
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/209564206-78edeb62-3239-433c-8053-4e12ac713d5f.MP4

https://user-images.githubusercontent.com/43996225/209564251-b630b890-ad3b-4b2a-88ce-1d0a0f428e74.MP4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @hungvu193 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1671962332211369

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Issue Owner</summary>Current Issue Owner: @mallenexpensify</details>

**Comment by mallenexpensify** [comment]
Unable to reproduce on iOS v1.2.42-2.
- Opened app
- Uploaded photo with internet - worked fine
- Went to airplane mode
- Uploaded photo - showed "Uploading attachment" in main chat
- Once connected back to internet, photo loaded once I went back into the chat. 

Tried to test on Android via Browserstack but was getting an endless spinner when trying to sign in.  

Anything I might be missing? @kavimuru can you please test again? Thanks

**Comment by MonilBhavsar** [comment]
I'm able to reproduce.

@mallenexpensify the internet needs to be turned off and then tap on image attachment

**Comment by MonilBhavsar** [comment]
On web also, there is no loader. But yeah we should stop showing infinite loader as the request already failed and display some message/icon

**Comment by mallenexpensify** [comment]
Thanks @MonilBhavsar . I was able to reproduce today (maybe I wasn't on the latest version ¯\_(ツ)_/¯ )
I'm unsure if this can be external so adding in an engineer....

**Comment by neil-marcellini** [comment]
We [decided](https://expensify.slack.com/archives/C049HHMV9SM/p1672766924935129?thread_ts=1671962332.211369&cid=C049HHMV9SM) to hold this until the Onyx image caching is working as described in the tracking issue here https://github.com/Expensify/App/issues/10894.

**Comment by mallenexpensify** [comment]
Looks like https://github.com/Expensify/App/issues/10894 is still open

**Comment by neil-marcellini** [comment]
Still on hold, but the latest update [here](https://github.com/Expensify/App/issues/10894#issuecomment-1396125418) is that the caching PR is very close.

**Comment by mallenexpensify** [comment]
- https://github.com/Expensify/App/issues/10894#issuecomment-1406630099 

> Quick little update: the React native web fix has been merged! I'm trying to publish our update (following [this SO](https://stackoverflowteams.com/c/expensify/questions/13855))
>
> Update: published here! https://www.npmjs.com/package/@expensify/react-native-web/v/0.18.11

Should be able to take off hold soon

**Comment by neil-marcellini** [comment]
I believe we are still waiting on the App PR for image caching based on the latest update https://github.com/Expensify/App/issues/10894#issuecomment-1425618847.

**Comment by neil-marcellini** [comment]
Still waiting.

**Comment by neil-marcellini** [comment]
Same, and it looks like we need some additional help on that issue. If I have free time I'll take a look.

**Comment by neil-marcellini** [comment]
I haven't had free time, but I'll try to take a look tomorrow and see if it's a good fit for me, or bump in Slack to see if anyone can work on the issue.

**Comment by neil-marcellini** [comment]
Still holding.

**Comment by neil-marcellini** [comment]
Holding, see the update [here](https://github.com/Expensify/App/issues/10894#issuecomment-1488503859).

**Comment by neil-marcellini** [comment]
I asked for a more clear update [here](https://github.com/Expensify/App/issues/10894#issuecomment-1515221675).

**Comment by neil-marcellini** [comment]
There's progress, but you better believe we're still holding

**Comment by neil-marcellini** [comment]
On hold forever? 😅

**Comment by mallenexpensify** [comment]
Still on hold
- https://github.com/Expensify/App/issues/10894

**Comment by neil-marcellini** [comment]
Same. That's a great way to post it, thanks Matt

**Comment by neil-marcellini** [comment]
Still on hold

**Comment by neil-marcellini** [comment]
[Apparently](https://github.com/Expensify/App/issues/10894#issuecomment-1842738899) the issue we're holding on is "super close"

**Comment by mallenexpensify** [comment]
Above is still open

**Comment by mallenexpensify** [comment]
hold

**Comment by neil-marcellini** [comment]
Still holding. There is some progress on the holding issue so that's good.

**Comment by MelvinBot** [comment]
This has been labelled "Needs Reproduction". Follow the steps here: https://stackoverflowteams.com/c/expensify/questions/16989

**Comment by mallenexpensify** [comment]
Unable to reproduce on iOS app.  Added retest-weekly to see if QA is able to repro.  If not, I'll close, since this is almost 1.5 years old 😨

**Comment by mvtglobally** [comment]
Issue is reproducible during KI retests.

No infinite loading, but also no error, just blank. 


https://github.com/Expensify/App/assets/43995119/68c73c1e-c275-4f7d-bdb6-15f7a4d8d549


https://github.com/Expensify/App/assets/43995119/f980045b-58f8-4121-99f3-b3eced397aef

**Comment by hungvu193** [comment]
It's being fixed here:

https://github.com/Expensify/App/issues/35502

**Comment by mallenexpensify** [comment]
PR for issue we're held on is actively being worked on.  Bumping this to weekly 
https://github.com/Expensify/App/issues/35502

**Comment by mallenexpensify** [comment]
PR this is held on is very, very close to merging
https://github.com/Expensify/App/pull/39290#issuecomment-2110701118

**Comment by mallenexpensify** [comment]
PR hit production two days ago 🎉 
- https://github.com/Expensify/App/pull/39290#issuecomment-2125960968

Unable to reproduce now, on Staging iOS. @neil-marcellini , should we **not** close this for some reason? 

https://github.com/Expensify/App/assets/22508304/76e164ba-8d29-4963-8eb0-4349f7c482a9

**Comment by hungvu193** [comment]
I think I'm eligible for reporting bonus for this one 😂. One of the first issue I reported here

**Comment by mallenexpensify** [comment]
Wow... looks like you might be.  Help me track @hungvu193 to make sure I have this right

- You reported the 'infinite spinner' bug ~forever ago~ on Dec 26th of 2022
- https://github.com/Expensify/App/issues/35502 was reported later and is very similar
- https://github.com/Expensify/App/pull/39290#issuecomment-2125960968 was the PR to fix #35502 and it also fixed this bug
- You're due compensation since you reported the bug first, even though it was another PR that fixed it (and the associated bug/issue for that was reported after). 

Sound right?

**Comment by hungvu193** [comment]
@mallenexpensify Yes. That's correct

**Comment by mallenexpensify** [comment]
Ok... Since the report was from 2022, I'm going to assume it was $250.  

@hungvu193 can you please accept the job and reply here once you have? 
https://www.upwork.com/jobs/~01c9e76cecfe58dcad

**Comment by hungvu193** [comment]
Sure. I've accepted. Thank you 😊

**Comment by mallenexpensify** [comment]
Bug Reporter: @hungvu193 paid $250 via Upwork.
